### ISSUE TITLE
Shellcheck -P option not working with $(dirname "${BASH_SOURCE[0]}")

### ISSUE THREAD

**Issue by juliandkatz** [koalaman/shellcheck#1644]
#### For bugs
- Rule Id: [SC1090](https://www.shellcheck.net/wiki/SC1090)
- My shellcheck version: docker hub's `koalaman/shellcheck:latest` as of date of this issue
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

set -euo pipefail

DIR=$(dirname "${BASH_SOURCE[0]}")
source "$DIR/debug.bash"

```

#### Here's what shellcheck currently says:
```
In test/setup.bash line 6:
source "$DIR/debug.bash"
       ^---------------^ SC1090: Can't follow non-constant source. Use a directive to specify location.
```

#### Here's what I wanted or expected to see:

No error, given my run command:

```
docker pull koalaman/shellcheck:latest && \
docker run --rm --tty -v "$PWD:/mnt" koalaman/shellcheck:latest -P SCRIPTDIR:test:. test/ignore.bash 
```

I believe `SCRIPTDIR` should be all that's necessary, but I tried including other options as well.

For context, both `debug.bash` is also in the `test` directory.

**Comment by reidpr** [comment]
Here is a reproducer for what I strongly suspect is the same issue:

```bash
#!/bin/bash

set -ex

mkdir -p /tmp/a /tmp/b

cat <<'EOF' > /tmp/a/a.sh
# shellcheck shell=sh

bdir=/tmp/b

. "${bdir}/b.sh"

true
EOF

cat <<'EOF' > /tmp/b/b.sh
# shellcheck shell=sh

true
EOF

SHELLCHECK=shellcheck
#SHELLCHECK='sudo docker run -it -v /tmp:/tmp koalaman/shellcheck:latest'

$SHELLCHECK -P /tmp/b -x /tmp/a/a.sh
```

Output:

```console
$ ~/shellcheck-bug.sh
+ mkdir -p /tmp/a /tmp/b
+ cat
+ cat
+ SHELLCHECK=shellcheck
+ shellcheck -P /tmp/b -x /tmp/a/a.sh

In /tmp/a/a.sh line 5:
. "${bdir}/b.sh"
  ^------------^ SC1090: Can't follow non-constant source. Use a directive to specify location.

For more information:
  https://www.shellcheck.net/wiki/SC1090 -- Can't follow non-constant source....
```

Expected output: No errors.

I see the issue on version 0.7.0 as well as the Docker image as of right now.

Thanks for your hard work on ShellCheck. It is a lifesaver for us!

**Comment by reidpr** [comment]
I am no longer able to reproduce this in ShellCheck 0.7.2 or the current Docker image.
